File: src/menu/components/icons/PlayerPropertyIcon.java
/*
	Klasse aller Icons die im ChoosePlayerPropertyMenu gelistet werden
*/

package menu.components.icons;

import player.property.PlayerProperty;
import graphics.ImageFile;
import graphics.ImageID;
import menu.components.Icon;
import menu.ComponentContainer;
import misc.math.pixel.PixelRect;
import misc.Debug;

public class PlayerPropertyIcon extends Icon
{
	public static final int WIDTH = 40;
	public static final int HEIGHT = 40;
	private PlayerProperty playerProperty;

	public PlayerPropertyIcon(ComponentContainer parent, PixelRect rect, PlayerProperty playerProperty)
	{
		super(parent, rect);
		this.playerProperty = playerProperty;
	}

	// Setter
	public void setPlayerProperty(PlayerProperty playerProperty)
	{
		this.playerProperty = playerProperty;
	}
	// getter
	public PlayerProperty getPlayerProperty() { return playerProperty; }
	@Override public ImageID getImageID()
	{
		// Falls noch kein 
		if (getPlayerProperty() == null)
		{
			Debug.warn("PlayerPropertyIcon.getImageID(): getPlayerProperty() == null -> returning ImageFile.VOID_ICON.getImageID()");
			return ImageFile.VOID_ICON.getImageID();
		}
		return getPlayerProperty().getIconImageID();
	}
}

File: cpp/src/menu/components/PlayerPropertyIcon.cpp
#include "PlayerPropertyIcon.hpp"

#include <playerproperty/PlayerProperty.hpp>

PlayerPropertyIcon::PlayerPropertyIcon(ComponentContainer* c, const PixelRect& r, PlayerProperty* p)
	: Icon(c, r)
{
	setPlayerProperty(p);
}

PlayerPropertyIcon::PlayerPropertyIcon(ComponentContainer* c, const PixelRect& r)
	: Icon(c, r)
{
	setPlayerProperty(NULL);
}

PixelVector PlayerPropertyIcon::getSize()
{
	return PixelVector(40, 40);
}

void PlayerPropertyIcon::setPlayerProperty(PlayerProperty* prop)
{
	property = prop;
}

TextureID PlayerPropertyIcon::getTextureID() const
{
	if (getPlayerProperty() == NULL || getPlayerProperty()->getIconTextureID() == -1)
	{
		return VOID_ICON_GID;
	}
	return getPlayerProperty()->getIconTextureID();
}

PlayerProperty* PlayerPropertyIcon::getPlayerProperty() const
{
	return property;
}
